src/pending_py_calls_notifier.h

```diff
@@ -0,0 +1,19 @@
+#ifndef PENDING_PY_CALLS_NOTIFIER_H
+#define PENDING_PY_CALLS_NOTIFIER_H
+
+#include <functional>
+
+namespace pending_py_calls_notifier {
+
+    // Initialize the notifier with a function that runs pending calls.
+    void initialize(std::function<void()> run_pending_calls);
+
+    // Notify the main thread to run pending calls.
+    void notify();
+
+    // Undo initialize
+    void deinitialize();
```

Comment:
Where does this get called?

Reply:
nowhere yet, still need to plumb it through to package unload.

Reply:
Is there a counterpart to `// [[Rcpp::init]]` but for finalize/unload?

Reply:
It is now called from `py_finalize()`, along with a bunch of other new finalizers.
Also, we now register `py_finalize()` on an object in the namespace, so the Python session is finalized on R session exit.